drivers/adc/ad7768/ad7768.c

```diff
@@ -626,21 +628,85 @@ int32_t ad7768_setup(ad7768_dev **device,
 	ad7768_dev *dev;
 	int32_t ret;
 
-	dev = (ad7768_dev *)no_os_malloc(sizeof(*dev));
+	ret = ad7768_setup_begin(&dev, init_param);
+	if (ret)
+		return ret;
+
+	ret = ad7768_setup_finish(dev, init_param);
+	if (ret)
+		goto error;
+
+	*device = dev;
+
+	return 0;
+error:
+	no_os_free(dev);
+
+	return -1;
```

Comment:
if you included `no_os_error.h` why not using the afferent error code?

Reply:
It is true. However, ret should be placed here.